enricher/fabric8/src/main/java/io/fabric8/maven/enricher/fabric8/DockerHealthCheckEnricher.java

```diff
@@ -0,0 +1,124 @@
+import io.fabric8.maven.core.util.KubernetesResourceUtil;
+import io.fabric8.maven.docker.config.HealthCheckConfiguration;
+import io.fabric8.maven.docker.config.HealthCheckMode;
+import io.fabric8.maven.docker.config.ImageConfiguration;
+import io.fabric8.maven.enricher.api.AbstractHealthCheckEnricher;
+import io.fabric8.maven.enricher.api.EnricherContext;
+
+import static io.fabric8.maven.enricher.api.util.GoTimeUtil.durationSeconds;
+
+/**
+ * Enrich a container with probes when health checks are defined in the {@code ImageConfiguration} of the docker maven plugin.
+ * This enricher could need a change when Dockerfile health checks will be supported natively in Openshift or Kubernetes.
+ */
+public class DockerHealthCheckEnricher extends AbstractHealthCheckEnricher {
+
+    public DockerHealthCheckEnricher(EnricherContext buildContext) {
+        super(buildContext, "docker-health-check");
+    }
+
+    /**
+     * Try to provide a health check for any image defined in the config.
+     */
+    @Override
+    protected List<Container> getCandidateContainers(Deployment deployment) {
+        if (deployment.getSpec() != null &&
+                deployment.getSpec().getTemplate() != null &&
+                deployment.getSpec().getTemplate().getSpec() != null) {
+
+            List<Container> containers = deployment.getSpec().getTemplate().getSpec().getContainers();
+            if (containers != null) {
+                return containers;
+            }
+        }
+
+        return Collections.emptyList();
+    }
+
+    @Override
+    protected Probe getReadinessProbe(Container container) {
+        return getProbe(container);
+    }
+
+    @Override
+    protected Probe getLivenessProbe(Container container) {
+        return getProbe(container);
+    }
+
+    protected Probe getProbe(ImageConfiguration image) {
+        if (hasHealthCheck(image)) {
+            HealthCheckConfiguration health = image.getBuildConfiguration().getHealthCheck();
+            return new ProbeBuilder()
+                    .withExec(new ExecAction(health.getCmd().asStrings()))
+                    .withTimeoutSeconds(durationSeconds(health.getTimeout()))
+                    .withPeriodSeconds(durationSeconds(health.getInterval()))
+                    .withFailureThreshold(health.getRetries())
+                    .build();
+        }
+
+        return null;
+    }
+
+    protected Probe getProbe(Container container) {
+        String imageName = container.getImage();
+        if (imageName == null) {
+            return null;
+        }
+
+        ImageConfiguration containerImage = null;
+        List<ImageConfiguration> images = getImages();
+        if (images != null) {
+            List<ImageConfiguration> candidate = new LinkedList<>();
+            for (ImageConfiguration image : images) {
+                if (imageName.equals(image.getName())) {
+                    candidate.add(image);
+                }
+            }
```

Comment:
I wonder whether there can be a situation where (a) this would not least to add a single image and (b) where more than one can be added.

the list of configured image come from the configured `<images>` plus the one generated by by generators. They all cary image names which are supposed to be unique (so only a single one can match). The `container.getImage()` comes from the `Deployment` object (or any other controller object) and should correspond to one of the configured images (90%). Ok, can't really imagine a use case where this refers to an externally build image, but this is also not forbidden. However when you don't find an image this way, there won't be one to examine for a healthcheck, so would stop here.